Content.Server/Xenoarchaeology/Equipment/ArtifactAnalyzerSystem.cs

```diff
@@ -31,21 +39,55 @@ private void OnExtractButtonPressed(Entity<AnalysisConsoleComponent> ent, ref An
         if (!_research.TryGetClientServer(ent, out var server, out var serverComponent))
             return;
 
+        // Begin DeltaV - Variables for extraction totals
         var sumResearch = 0;
+        var sumGlimmer = 0;
+        if (!_analyzerSystem.TryGetAnalyzer(ent, out var analyzer))
+            return;
+        // End DeltaV
+
         foreach (var node in _xenoArtifact.GetAllNodes(artifact.Value))
         {
             var research = _xenoArtifact.GetResearchValue(node);
             _xenoArtifact.SetConsumedResearchValue(node, node.Comp.ConsumedResearchValue + research);
+
+            // Begin DeltaV - Only run if we have an artifact ready for extraction
+            if (analyzer != null)
+            {
+                sumGlimmer += (int)(research / (float)analyzer.Value.Comp.ExtractRatio);
+                research = (int)(research * GetGlimmerMultiplier(analyzer.Value.Comp));
+            }
+            // End DeltaV
             sumResearch += research;
         }
+        if (analyzer != null)
+        {
+            UpdateClientUI(ent, analyzer);
+        }
```

Comment:
Just need a deltav comment, but we can also simplify now that analyzer is never null, if you want
```suggestion
        UpdateClientUI(ent, analyzer); // DeltaV
```

Reply:
> Just need a deltav comment, but we can also simplify now that analyzer is never null, if you want

I had to tweak this change I think it should be good now

Reply:
Weird, I wonder why compiler wasn't happy with that

Reply:
I actually think it was an earlier change and I didnt catch the issue until now. It didnt even stop my build from running until I refreshed VS